### PR TITLE
adding julia image

### PR DESCRIPTION
**Pull request by nathanballou**
Adds a Julia image called saturnbase-julia containing Julia 1.7.2. This mostly copies saturnbase, but includes additional code to download Julia, link it, and install IJulia for Jupyter integration. 

It attempts to be as lightweight as possible, but there may be additional packages we can remove etc.

The image also contains python 3.9 as an alternative kernel - we might want to remove it.

You install Julia packages using a start script like the following. Perhaps another place for improvement.
```
julia -e 'import Pkg;
Pkg.add("Package1");
Pkg.add("Package2")'
```
 or 

```
julia -e 'import Pkg;
Pkg.add([
"Package1",
"Package2"
])'
```

### CODE REVIEW

**Comment by nathanballou** [comment]
Added saturn-julia containing the following packages. We can refine over time.
Convex
DataFrames
Dierckx
ExcelReaders
Flux
Gadfly
HDF5
Ipopt
JLD
JuMP
LsqFit
MLBase
NMF
Optim
Plots
ProgressMeter
PyCall
PyPlot
RDatasets
SCS
SQLite
Spectra
Turing

**Comment by nathanballou** [comment]
"Bio",
"Clustering",
"DataFrames", 
"Distances",
"Flux", 
"HDF5",
"MLBase",
"Optim",
"Plots",
"ProgressMeter",
"PyCall",
"PyPlot",
"RDatasets",
"SCS",
"Statistics",
"SQLite"

**Comment by forana** [comment]
We aren't planning on building any new versions of `release-2022.01.06`, as far as I know - this should merge into either `release-2022.03.01` or `main`.

File: saturn-julia/.env_deps [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VERSION=2022.01.06-1
```
Comment (jnolis) [inline saturn-julia/.env_deps:1]: I don't think this file is actually used for anything, but the dates are old

File: saturn-julia/Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+include .env_deps
+export
+
+build_image:
+	docker build \
+		--no-cache \
+		--build-arg SATURNBASE_IMAGE=${SATURNBASE_IMAGE} \
+		-t narobboo/julia-base:0.06 \
```
Comment (jnolis) [inline saturn-julia/Makefile:8]: what is narobboo??

File: saturn-julia/recipe-template.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "saturn-julia",
+    "description": "A Julia image for Saturn Cloud. Contains Julia and Python along with a selection of popular Julia packages.",
+    "hardware_type": "cpu",
+    "supports": [
+        "jupyterlab",
+        "julia"
```
Comment (jnolis) [inline saturn-julia/recipe-template.json:7]: Did alex ask for this? I'm surprised that we're adding a supports

File: saturnbase-julia/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    sudo chown "${NB_USER}" "${JULIA_PKGDIR}" && \
+    fix-permissions "${JULIA_PKGDIR}" && \
+    # Install IJulia as jovyan
+    julia -e 'import Pkg; Pkg.update()' && \
+    julia -e 'using Pkg; pkg"add IJulia"; pkg"precompile"' && \
+    # move kernelspec out of system share
+    mv "${HOME}/.local/share/jupyter/kernels/julia"* "${CONDA_DIR}/share/jupyter/kernels/" && \
+    chmod -R go+rx "${CONDA_DIR}/share/jupyter" && \
+    rm -rf "${HOME}/.local" && \
+    fix-permissions "${JULIA_PKGDIR}" "${CONDA_DIR}/share/jupyter"
```
Comment (jnolis) [inline saturnbase-julia/Dockerfile:135]: Should you delete the fix-permissions script at the end?

**Review by jnolis — CHANGES_REQUESTED** [review]

**Review by jnolis — APPROVED** [review]
Love it!

**Pull request merged**